cmd/gpu_fakedev/gpu_fakedev.go

```diff
@@ -291,6 +296,88 @@ func generateDriFiles(opts genOptions) {
 	log.Printf("Done, created %d dirs, %d devices and %d files.", opts.dirs, opts.devs, opts.files)
 }
 
+func makeXelinkSideCar(topology string, gpus, tiles int, connections string) {
+	if topology != fullyConnected {
+		log.Printf("XELINK: generate xelink sidecar label file, using (GPUs: %d, Tiles: %d)", gpus, tiles)
+	} else {
+		log.Printf("XELINK: generate xelink sidecar label file, using (GPUs: %d, Tiles: %d, Topology: %s)", gpus, tiles, topology)
+	}
+
+	if topology == fullyConnected {
+		saveSideCarFile(buildConnectionList(gpus, tiles))
+	} else {
+		saveSideCarFile(connections)
+	}
```

Comment:
Simpler to have just one single comparison, instead of two opposite ones:
```suggestion
	if topology != fullyConnected {
		log.Printf("XELINK: generate xelink sidecar label file, using (GPUs: %d, Tiles: %d)", gpus, tiles)
		saveSideCarFile(connections)
	} else {
		log.Printf("XELINK: generate xelink sidecar label file, using (GPUs: %d, Tiles: %d, Topology: %s)", gpus, tiles, topology)
		saveSideCarFile(buildConnectionList(gpus, tiles))
	}
```